Examples/ObjectiveC/ClusteringExample.m

```diff
@@ -0,0 +1,138 @@
+    self.mapView.autoresizingMask = UIViewAutoresizingFlexibleWidth | UIViewAutoresizingFlexibleHeight;
+    self.mapView.delegate = self;
+    [self.view addSubview:self.mapView];
+
+    self.sprite = [UIImage imageNamed:@"sprite"];
+}
+
+- (void)mapView:(MGLMapView *)mapView didFinishLoadingStyle:(MGLStyle *)style {
+    NSURL *url = [NSURL fileURLWithPath:[[NSBundle mainBundle] pathForResource:@"ports" ofType:@"geojson"]];
+
+    MGLShapeSource *source = [[MGLShapeSource alloc] initWithIdentifier:@"clusteredPorts"
+                                                                    URL:url
+                                                                options:@{ MGLShapeSourceOptionClustered: @(YES),
+                                                                           MGLShapeSourceOptionClusterRadius: @(self.sprite.size.width) }];
+    [style addSource:source];
+
+    // Use a template image so that we can tint it with the `iconColor` runtime styling property.
+    [style setImage:[self.sprite imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate] forName:@"sprite"];
+
+    // Show unclustered features as icons. The `cluster` attribute is built into clustering-enabled source features.
+    MGLSymbolStyleLayer *ports = [[MGLSymbolStyleLayer alloc] initWithIdentifier:@"ports" source:source];
+    ports.iconImageName = [MGLStyleValue valueWithRawValue:@"sprite"];
+    ports.iconColor = [MGLStyleValue valueWithRawValue:[[UIColor darkGrayColor] colorWithAlphaComponent:0.9]];
+    ports.predicate = [NSPredicate predicateWithFormat:@"%K != YES", @"cluster"];
+    [style addLayer:ports];
+
+    // Color clustered features based on clustered point counts.
+    NSDictionary *stops = @{ @20:  [MGLStyleValue valueWithRawValue:[UIColor lightGrayColor]],
+                             @50:  [MGLStyleValue valueWithRawValue:[UIColor orangeColor]],
+                             @100: [MGLStyleValue valueWithRawValue:[UIColor redColor]],
+                             @200: [MGLStyleValue valueWithRawValue:[UIColor purpleColor]] };
+
+    // Show clustered features as circles. The `point_count` attribute is built into clustering-enabled source features.
+    MGLCircleStyleLayer *circlesLayer = [[MGLCircleStyleLayer alloc] initWithIdentifier:@"clusteredPorts" source:source];
+    circlesLayer.circleRadius = [MGLStyleValue valueWithRawValue:@(self.sprite.size.width / 2)];
+    circlesLayer.circleOpacity = [MGLStyleValue valueWithRawValue:@0.75];
+    circlesLayer.circleStrokeColor = [MGLStyleValue valueWithRawValue:[[UIColor whiteColor] colorWithAlphaComponent:0.75]];
+    circlesLayer.circleStrokeWidth = [MGLStyleValue valueWithRawValue:@2];
+    circlesLayer.circleColor = [MGLSourceStyleFunction functionWithInterpolationMode:MGLInterpolationModeInterval
+                                                                              stops:stops
```

Comment:
With the current width limitations of the /ios-sdk site, we can’t use normal ObjC indentation because it ends up looking like this:

<img width="766" alt="screen shot 2017-03-01 at 11 21 30 am" src="https://cloud.githubusercontent.com/assets/1198851/23477072/aa61a65e-fe71-11e6-81c1-81eb12309078.png">